### PR TITLE
[Feat] delete user account

### PR DESCRIPTION
**Pull request by kilemensi**
## Description

Allow users to delete their account (including `SocialAccount` if signed in via Google) by sending `DELETE` request to `/auth/user` end point.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshots

N/A

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: twoopstracker/authentication/views.py [hunk 1]
```diff
@@ -1,9 +1,30 @@
...
+
+class UserDetailsView(BaseUserDetailsView, RetrieveUpdateDestroyAPIView):
+    def perform_destroy(self, instance):
+        social_account = SocialAccount.objects.filter(user=instance).first()
+        if social_account:
+            # NOTE(kilemensi): Since we're deleting it, skip any validation
+            #                  such as
+            # https://github.com/pennersr/django-allauth/blob/0.47.0/allauth/socialaccount/adapter.py
+            social_account.delete()
+            signals.social_account_removed.send(
```
Comment (esirK) [inline twoopstracker/authentication/views.py:24]: Given that we don't have any receiver for this signal, why are we sending it?

Comment (kilemensi) [reply twoopstracker/authentication/views.py:24]: That's how [**dj-rest-auth**](https://github.com/iMerica/dj-rest-auth) disconnects a [`SocialAccount`](https://github.com/iMerica/dj-rest-auth/blob/2d455333d5ae9910adb43de3488013ee2028f2eb/dj_rest_auth/registration/views.py#L216-L221).

Comment (esirK) [reply twoopstracker/authentication/views.py:24]: yes but for our case, unless we want to perform a certain action when a social account is deleted, I don't think we need to send out the signal.

**Review by esirK — APPROVED** [review]
🚀

File: twoopstracker/authentication/views.py (hunk 1 above)
Comment (kilemensi) [reply twoopstracker/authentication/views.py:24]: Why? For example Django sends signals whenever a model is saved, deleted, etc., and 99% of the the time, we don't use them. Why aren't we disabling them?

Unless there is a reason to break the expectation set by dj_rest_auth, P[rinciple of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) should apply

**Pull request merged**